pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -264,70 +310,49 @@ func removeSSHRule(ctx context.Context, cl k8sClient.Client, infraID, region str
+		awsCluster.Spec.NetworkSpec.AdditionalControlPlaneIngressRules = postBootstrapRules
 
-	var sgID string
-	if sg, ok := awsCluster.Status.Network.SecurityGroups[capa.SecurityGroupControlPlane]; ok && len(sg.ID) > 0 {
-		sgID = sg.ID
-	} else if ok {
-		return fmt.Errorf("control plane security group id is not populated in awscluster status")
-	} else {
-		keys := make([]capa.SecurityGroupRole, 0, len(awsCluster.Status.Network.SecurityGroups))
-		for sgr := range awsCluster.Status.Network.SecurityGroups {
-			keys = append(keys, sgr)
+		if err := cl.Update(ctx, awsCluster); err != nil {
+			return fmt.Errorf("failed to update AWSCluster during bootstrap destroy: %w", err)
 		}
-		return fmt.Errorf("controlplane not found in cluster security groups: %v", keys)
+		logrus.Debug("Updated AWSCluster to remove bootstrap SSH rule")
 	}
 
-	timeout := 15 * time.Minute
-	startTime := time.Now()
-	untilTime := startTime.Add(timeout)
-	timezone, _ := untilTime.Zone()
-	logrus.Debugf("Waiting up to %v (until %v %s) for bootstrap SSH rule to be destroyed...", timeout, untilTime.Format(time.Kitchen), timezone)
-	if err := wait.PollUntilContextTimeout(ctx, 15*time.Second, timeout, true,
-		func(ctx context.Context) (bool, error) {
-			sgs, err := awsconfig.DescribeSecurityGroups(ctx, session, []string{sgID}, region)
-			if err != nil {
-				return false, fmt.Errorf("error getting security group: %w", err)
-			}
+	return nil
+}
 
-			if len(sgs) != 1 {
-				ids := []string{}
-				for _, sg := range sgs {
-					ids = append(ids, *sg.GroupId)
-				}
-				return false, fmt.Errorf("expected exactly one security group with id %s, but got %v", sgID, ids)
-			}
+// isSSHRuleGone checks that the Public SSH rule has been removed from the security group.
+func isSSHRuleGone(ctx context.Context, session *session.Session, region, sgID string) (bool, error) {
+	sgs, err := awsconfig.DescribeSecurityGroups(ctx, session, []string{sgID}, region)
+	if err != nil {
+		return false, fmt.Errorf("error getting security group: %w", err)
+	}
 
-			sg := sgs[0]
-			publicSSHRulesFound := false
-			for _, rule := range sg.IpPermissions {
-				if ptr.Deref(rule.ToPort, 0) != 22 {
-					continue
-				}
-				for _, source := range rule.IpRanges {
-					if source.CidrIp != nil && *source.CidrIp == "0.0.0.0/0" {
-						publicSSHRulesFound = true
-						ruleDesc := ptr.Deref(source.Description, "[no description]")
-						logrus.Debugf("Found security group %s with source cidr %s. Still waiting for deletion...", ruleDesc, *source.CidrIp)
-					}
-				}
+	if len(sgs) != 1 {
+		ids := []string{}
+		for _, sg := range sgs {
+			ids = append(ids, *sg.GroupId)
+		}
+		return false, fmt.Errorf("expected exactly one security group with id %s, but got %v", sgID, ids)
+	}
+
+	sg := sgs[0]
+	for _, rule := range sg.IpPermissions {
+		if ptr.Deref(rule.ToPort, 0) != 22 {
+			continue
+		}
+		for _, source := range rule.IpRanges {
+			if source.CidrIp != nil && *source.CidrIp == "0.0.0.0/0" {
```

Comment:
@patrickdillon because of the comparison to the allow-all cidr, we'll never delete the ssh rule for private clusters: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/aws/cluster.go#L33-L36.
Is that an issue?